src/module/dy.js

```diff
@@ -33,20 +33,34 @@ module.exports = ({ call, getService, logger }) => ({
     };
     return ({
       upsert: async (item, {
-        conditions = null
+        conditions = null,
+        expectedErrorCodes = []
       } = {}) => {
-        const result = await model.entity.update(item, {
-          returnValues: 'all_old',
-          ...(conditions === null ? {} : { conditions })
-        });
+        assert(Array.isArray(expectedErrorCodes));
+        let result;
+        try {
+          result = await model.entity.update(item, {
+            returnValues: 'all_old',
+            ...(conditions === null ? {} : { conditions })
+          });
+        } catch (err) {
+          if (expectedErrorCodes.includes(err.code)) {
+            return err.code;
+          }
+          throw err;
+        }
         const created = result.Attributes === undefined;
         await (created === true ? onCreate : onUpdate)(item);
         return { created };
       },
       update: async (item, {
         returnValues = 'all_new',
-        conditions: updateConditions = null
+        conditions: updateConditions = null,
+        onItemNotFound = onNotFound,
```

Comment:
```suggestion
        onNotFound = onNotFound_,
```
why don't we do that? I think it's confusing to have them different?